src/Microsoft.VisualStudio.Editor.Razor/TextBufferExtensions.cs

```diff
@@ -17,5 +20,34 @@ public static bool IsRazorBuffer(this ITextBuffer textBuffer)
 
             return textBuffer.ContentType.IsOfType(RazorLanguage.ContentType);
         }
+
+        public static VisualStudioDocumentTracker GetDocumentTracker(this ITextBuffer textBuffer)
```

Comment:
Why do we need this? Can you just use the service?

Reply:
I really really don't like the idea of hanging lots of interdependent functionality on extensions, I'd prefer that it use services.

For instance you could add `GetFocusedTextView` to the document tracker service

Reply:
Sure we can move these